### PR TITLE
Implement parsing git-repo URLs from POMs

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/958)

### CODE REVIEW

File: analyzer/src/test/kotlin/MavenSupportTest.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import org.apache.maven.model.Scm
+import org.apache.maven.project.MavenProject
+
+class MavenSupportTest : WordSpec({
+    "parseVcsInfo()" should {
+        "handle GitRepo URLs" {
+            val mavenProject = MavenProject().apply {
+                scm = Scm().apply {
+                    connection = "scm:git-repo:ssh://host.com/project/foo?path/to/manifest.xml"
```
Comment (mnonnenmacher) [inline analyzer/src/test/kotlin/MavenSupportTest.kt]: Is this a documented way to write Git-Repo URLs, or is it something we made up?

File: analyzer/src/main/kotlin/MavenSupport.kt [hunk 2]
```diff
@@ -143,28 +143,28 @@ class MavenSupport(workspaceReader: WorkspaceReader) {
...
                     val url = it.group("url")
 
                     // CVS URLs usually start with ":pserver:" or ":ext:", but as ":" is also the delimiter used by the
                     // Maven SCM plugin, no double ":" is used in the connection string and we need to fix it up here.
-                    if (type == "cvs" && !url.startsWith(":")) {
-                        Pair(type, ":$url")
-                    } else {
-                        Pair(type, url)
+                    when {
+                        type == "cvs" && !url.startsWith(":") -> listOf(type, ":$url", "")
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/MavenSupport.kt]: The comment from before `when` should now go before this line, and probably another comment should be added before the line below, explaining the substrings used.

File: analyzer/src/main/kotlin/MavenSupport.kt [hunk 3]
```diff
@@ -143,28 +143,28 @@ class MavenSupport(workspaceReader: WorkspaceReader) {
...
                         log.info { "Ignoring Maven SCM connection URL '$connection' of unexpected format." }
                     }
 
-                    Pair("", "")
+                    listOf("", "", "")
                 }
             }
 
-            return VcsInfo(type, url, tag)
+            return VcsInfo(type, url, tag, null, path)
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/MavenSupport.kt]: Better name the last argument than provide `null`.

File: analyzer/src/test/kotlin/MavenSupportTest.kt (hunk 1 above)
Comment (sschuberth) [reply analyzer/src/test/kotlin/MavenSupportTest.kt]: It's something that I made up to work around a concrete issue that we currently have. See http://maven.apache.org/scm/scms-overview.html for the list of officially supported SCMs.

File: analyzer/src/main/kotlin/MavenSupport.kt (hunk 2 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/MavenSupport.kt]: Done.

File: analyzer/src/main/kotlin/MavenSupport.kt (hunk 3 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/MavenSupport.kt]: Done.

File: analyzer/src/test/kotlin/MavenSupportTest.kt (hunk 1 above)
Comment (sschuberth) [reply analyzer/src/test/kotlin/MavenSupportTest.kt]: Let's discuss this separately when it's more clear if we need to implement case 2), which I'd rather not support, as for me git-repo manifests simply belong into Git repos.

Comment (fviernau) [reply analyzer/src/test/kotlin/MavenSupportTest.kt]: Agree, makes sense.

**Pull request merged**